Enable all cudas

martinwicke opened the pull request.

vrv commented on configure:
```diff
@@ -105,23 +94,27 @@ while true; do
     exit 1
   fi
-  TF_CUDNN_VERSION=""
+TF  TF_CUDNN_VERSION=""
```
is this a typo?

martinwicke replied:
I wonder why the tests pass.

vrv commented on configure:
```diff
@@ -202,10 +195,8 @@ EOF
-if [ "$TF_UNOFFICIAL_SETTING" == "1" ]; then
-  UnofficialSetting
-fi
+# Ask for the unofficial settings
```
Maybe we should just say 'ask for user-configuration' and rename the function

vrv commented on tensorflow/core/platform/default/build_config.bzl:
```diff
@@ -4,8 +4,8 @@ load("//google/protobuf:protobuf.bzl", "cc_proto_library")
 # configure may change the following lines.
-CUDA_VERSION = '7.0'
-CUDNN_VERSION = '6.5'
+CUDA_VERSION = '.7.0'
```
1) isn't configure *going* to change these lines?

2) Should the default be empty, or at least, default to the latest version we know to work?

martinwicke replied:
So far, that is the latest version we (officially) know will work.

martinwicke replied:
Ok. I'll change it to empty. You're right.

martinwicke commented:
@zheng-xq, @vrv, tell me what you think.

vrv commented on configure:
```diff
@@ -129,22 +122,22 @@ CUDA_TOOLKIT_PATH="$CUDA_TOOLKIT_PATH"
   # Configure the Cuda toolkit version to work with.
-  perl -pi -e "s,CUDA_VERSION = '[0-9\.]*',CUDA_VERSION = '$TF_CUDA_VERSION',s" tensorflow/core/platform/default/build_config.bzl
-  perl -pi -e "s,(GetCudaVersion.*return )\"[0-9\.]*\",\1\"$TF_CUDA_VERSION\",s" tensorflow/stream_executor/dso_loader.cc
+  perl -pi -e "s,CUDA_VERSION = '[0-9\.]*',CUDA_VERSION = '.$TF_CUDA_VERSION',s" tensorflow/core/platform/default/build_config.bzl
```
These two lines might go in a separate function, since it's not 'unofficial'

Also, line 176 probably should be removed....

martinwicke commented:
I messed this up, so I'll have to fix the merge. :(

vrv commented:
1) yay for building!

2) So as written, we now use the system default as the default, even in Docker/Jenkins, correct?  So we would need to just install cudnn R4 on our docker instance and we'll start testing that version?

I think this looks good to me, but still want @zheng-xq to take a look :)

vrv commented:
Also, you might want to squash the commits at this point so the diff is easier to read.  Sigh GitHub's code review tools.

martinwicke commented:
Yes, we can either just install R4 to test, or we set the version explicitly using TF_CUDNN_VERSION.

martinwicke committed:
Simplify choosing different Cuda/cuDNN versions

Makes it easier to select a different version of Cuda or cuDNN. The configure
script will now always ask which version of Cuda and cuDNN is to be used if
GPU support is requested. Leaving the versions empty will build against the
system's default version.

martinwicke commented:
Squashed.

martinwicke commented:
Jenkins, test this please. I merged the fixes to the GPU pip build script so you have to test it again.

martinwicke commented:
Jenkins, test this again, please.

martinwicke commented:
Jenkins, test this please. Some permissions were broken in the android build. Not sure why.

martinwicke commented:
merged.
